### PR TITLE
feat: add MUSA out-of-tree vendor backend with op implementations

### PR DESCRIPTION
**Pull request by huangxu0914**
## Summary

Add out-of-tree (OOT) vendor backend for Moore Threads MUSA platform,
including dispatcher, op registrations, and initial op kernel implementations.

## Background

SGLang's plugin dispatch system supports multiple backend tiers
(flagos > vendor > reference). This PR adds `vendor.musa` (priority=100)
as the MUSA-specific tier, sitting between flagos (150) and reference (50).
Ops that lack a native MUSA kernel raise `NotImplementedError` and
automatically fall back to the next tier.

### vendor/musa backend
- `musa.py`: `MUSABackend` dispatcher
- `register_ops.py`: registers 10 ops
- Op implementations:
  | Op | Implementation |
  |---|---|
  | `silu_and_mul` | pure torch silu×mul, mirrors sglang `forward_musa` |
  | `rms_norm` | `sgl_kernel.fused_add_rmsnorm` / `nn.functional.rms_norm` |
  | `gemma_rms_norm` | `sgl_kernel.gemma_rmsnorm` / `gemma_fused_add_rmsnorm` |
  | `fused_moe` | delegates to `forward_cuda` (sglang official MUSA path) |
  | `topk` | sglang upstream MUSA Triton kernels (`topk_softmax`/`topk_sigmoid`) |
  | `rotary_embedding` / `mrotary_embedding` / FLA ops | `NotImplementedError` → fallback |

### Reference backend additions
- `reference/impl/fla.py`: three FLA ops via fla-project Triton kernels
- `reference/impl/fused_moe.py`: pure PyTorch fused MoE reference
- `reference/impl/mrotary_embedding.py`: rewrite to mirror sglang
  `forward_native` — merge cos/sin per `mrope_section` axis first,
  then apply standard RoPE

### MUSA dispatch fix
- `dispatch/rotary_patch.py`: MUSA overwrites `_forward_method =
  forward_native` in `RotaryEmbedding.__init__`, bypassing our bridge hook.
  This patch restores the hook after init.

### Platform support
- `dispatch/config/musa.yaml`, `platform.py`, `config/utils.py`:
  MUSA platform detection and config loading

### Examples
- `page_size=1` injected only on MUSA (workaround for sglang MUSA bug);
  other platforms use the sglang default

## Test

Validated on Moore Threads S5000 with two models:

| Model | Key ops dispatched |
|---|---|
| Qwen3.6-27B (Dense) | gemma_rms_norm→vendor.musa, silu_and_mul→flagos, mrotary_embedding→reference.torch, FLA→reference.triton |
| Qwen3.6-35B-A3B (MoE) | + topk→vendor.musa (flagos bug fallback), fused_moe→vendor.musa |

Both inference examples pass validation (`All validations passed`).

### CODE REVIEW

**Comment by physics31415926** [comment]
LGTM

File: sglang_fl/dispatch/backends/reference/impl/fused_moe.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+
+def fused_moe_torch(
+    obj,
+    layer: torch.nn.Module,
+    dispatch_output,
+):
+    from sglang.srt.layers.moe.moe_runner.triton import TritonMoeQuantInfo
+
+    quant_info = TritonMoeQuantInfo(
```
Comment (yixiaodapeng) [inline sglang_fl/dispatch/backends/reference/impl/fused_moe.py]: put triton implementation into vendor musa

File: sglang_fl/dispatch/backends/vendor/musa/impl/activation.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+#   which uses nn.SwishGLU — pure torch_musa ops, no Triton required.
+
+from __future__ import annotations
+
+import torch
+
+
+def silu_and_mul_musa(obj, x: torch.Tensor) -> torch.Tensor:
+    d = x.shape[-1] // 2
+    return torch.nn.functional.silu(x[..., :d]) * x[..., d:]
```
Comment (yixiaodapeng) [inline sglang_fl/dispatch/backends/vendor/musa/impl/activation.py]: reference implementation part in vendor？if lacking vendor implementation，raise NOT IMPLEMENTATION

**Review by physics31415926 — APPROVED** [review]

File: sglang_fl/dispatch/backends/vendor/musa/impl/fused_moe.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+import torch
+
+
+def fused_moe_musa(
+    obj,
+    layer: torch.nn.Module,
+    dispatch_output,
+):
+    return obj.forward_cuda(layer, dispatch_output)
```
Comment (yixiaodapeng) [inline sglang_fl/dispatch/backends/vendor/musa/impl/fused_moe.py]: use forward_musa

**Review by yixiaodapeng — APPROVED** [review]

**Review by aoyulong — APPROVED** [review]
